.circleci/config.yml

```diff
@@ -139,6 +143,13 @@ references:
   store_test_logs: &store_test_logs
     store_artifacts:
         path: log/test
+  report_code_coverage: &report_code_coverage
+    run:
+        name: Report Code Coverage
+        command: |
+          export ci_env=`bash <(curl -s https://codecov.io/env)`
+          . install/setup.sh
+          src/navigation2/tools/code_coverage_report.bash codecovio
```

Comment:
I feel like codecov's orb could achieve this, but I don't yet know a lot about lcov. 
https://circleci.com/orbs/registry/orb/codecov/codecov

Reply:
lcov fundamentally just gathers up all the statistics files generated by the test run and consolidates them into a single database. The extra commands are to make sure no files are missed and to remove files from the report that I don't care about (eg system files and generated message files).

The orb seems to basically just do that one line `bash <(curl -s https://codecov.io/env)`. I'm not sure if it is worth it.

Reply:
Where are we passing/consuming the `CODECOV_TOKEN`?

Reply:
Your right, the orb is just a wrapper around the upload, nevermind.